docs/_docs/2_2_writing_data.md

```diff
@@ -176,15 +176,49 @@ In some cases, you may want to migrate your existing table into Hudi beforehand.
 
 ## Datasource Writer
 
-The `hudi-spark` module offers the DataSource API to write (and also read) any data frame into a Hudi table.
-Following is how we can upsert a dataframe, while specifying the field names that need to be used
-for `recordKey => _row_key`, `partitionPath => partition` and `precombineKey => timestamp`
+The `hudi-spark` module offers the DataSource API to write (and read) a Spark DataFrame into a Hudi table. There are a number of options available:
 
+**`HoodieWriteConfig`**:
+
+**TABLE_NAME** (Required)<br>
+
+
+**`DataSourceWriteOptions`**:
+
+**RECORDKEY_FIELD_OPT_KEY** (Required): Primary key field(s). Nested fields can be specified using the dot notation eg: `a.b.c`. When using multiple columns as primary key use comma seperated notaion, eg: `"col1,col2,col3,etc"`. Single or multiple columns as primary key specified by `KEYGENERATOR_CLASS_OPT_KEY` property.<br>
+Default value: `"uuid"`<br>
+
+**PARTITIONPATH_FIELD_OPT_KEY** (Required): Columns to be used for partitioning the table. To prevent partitioning, provide empty string as value eg: `""`. Specify paritioning/no partitioning using `HIVE_PARTITION_EXTRACTOR_CLASS_OPT_KEY`<br>
```

Comment:
`HIVE_PARTITION_EXTRACTOR_CLASS_OPT_KEY` should be changed to `KEYGENERATOR_CLASS_OPT_KEY`? and also when specify no partitioning tables, would use `NonpartitionedKeyGenerator`

Reply:
Yap, seems both are necessary:
```suggestion
**PARTITIONPATH_FIELD_OPT_KEY** (Required): Columns to be used for partitioning the table. To prevent partitioning, provide empty string as value eg: `""`. Specify partitioning/no partitioning using both `KEYGENERATOR_CLASS_OPT_KEY` and `HIVE_PARTITION_EXTRACTOR_CLASS_OPT_KEY`<br>
```

Reply:
> Yap, seems both are necessary:

The `HIVE_PARTITION_EXTRACTOR_CLASS_OPT_KEY` is used to sync to hive, if you are meaning sync to hive during the write, it makes sense, or it makes no sense.

Reply:
I see under the "syncing to hive" section it states "Both tools above support syncing of the table’s latest schema to Hive metastore" which implies that syncing table metadata to hive is not mandatory but optional. Is this actually true? It seems that using a metastore is actually mandatory...

Reply:
One would only write data into hudi but not syncing to hive, so the `HIVE_PARTITION_EXTRACTOR_CLASS_OPT_KEY ` is only used when syncing to hive metastore.

Reply:
Will add clarification that HIVE_PARTITION_EXTRACTOR_CLASS_OPT_KEY is only necessary when using hive.